sdk/src/azure/core/az_mqtt5_rpc_server_hfsm.c

```diff
@@ -236,8 +239,7 @@ AZ_INLINE az_result _handle_request(
   az_span correlation_data_bindata = az_mqtt5_property_get_binarydata(&correlation_data);
   az_span content_type_str = az_mqtt5_property_get_string(&content_type);
 
-  if (az_span_ptr(response_topic_str) != NULL && az_span_ptr(correlation_data_bindata) != NULL
-      && az_span_ptr(content_type_str) != NULL)
+  if (az_span_ptr(response_topic_str) != NULL && az_span_ptr(correlation_data_bindata) != NULL)
```

Comment:
@vaavva I think we should reconsider all these checks and allow the RPC HFSMs to make progress as much as they can based on the information available.
We could log warnings in the case where the protocol is broken but the information can still be relayed to the application as incomplete (`AZ_SPAN_EMPTY` for missing fields).

Reply:
I'll add another PR for that.